### PR TITLE
Log QUIC_TLS_SECRETS on all TestConnections and write to disk on cleanup

### PR DESCRIPTION
**Pull request by anrossi**
## Description

Logs QUIC_TLS_SECRETs on all TestConnections. If a filepath is provided, then it will write the QUIC_TLS_SECRETS in an sslkeylogfile format to the provided file path when the TestConnection is destructed.

## Testing

CI

## Documentation

N/A

### CODE REVIEW

File: src/test/lib/TestConnection.h [hunk 1]
```diff
@@ -17,6 +17,8 @@ enum NEW_STREAM_START_TYPE {
     NEW_STREAM_START_ASYNC      // Start asynchronously
 };
 
+#define DEFAULT_SSLKEYLOGFILE_PATH "sslkeylogfile.txt"
```
Comment (ami-GS) [inline src/test/lib/TestConnection.h:20]: PATH? not FILENAME?

Comment (anrossi) [reply src/test/lib/TestConnection.h:20]: I don't know what word to use to describe that it's both path and filename without writing that out and making the name longer

**Comment by anrossi** [comment]
DO NOT COMPLETE THIS PR. Trying out using a different `SetParam` in the client TestConnection constructor to see if it's a SetParam issue, or a TLS_SECRETS issue.

**Comment by anrossi** [comment]
I can't repro the test failure locally.

**Review by nibanks — APPROVED** [review]
LGTM!

**Pull request merged**